README.md

```diff
@@ -41,6 +40,42 @@ and whilst rails is an excellent tool, for smaller projects you may want to look
 as sinatra etc...
 
 
+### Client IP address tracing in Rails apps
+
+
+Add the request store gem to your Gemfile and install
+
+```ruby
+gem "request-store"
+```
+
+```sh
+bundle install
+```
+
+```ruby
+# application_controller.rb
+class ApplicationController
+  before_action :set_request_ip
+
+  private
+
+  def set_request_ip
+    RequestStore[:request_ip] = request.ip
```

Comment:
Would be better to recommend `remote_ip` here:

```suggestion
    RequestStore[:request_ip] = request.remote_ip
```

This is the IP address set by the `RemoteIp` middleware. This middleware goes through a list of potential headers, removing any common proxy IPs and then grabs what it thinks is the "most likely" originating IP address. https://github.com/rails/rails/blob/5-2-stable/actionpack/lib/action_dispatch/middleware/remote_ip.rb

Reply:
Based on this comment it might make literal sense to rename the header as 'X-Forwarded-Remote-for' or something that matches the headers implementation in rails with its name inside the HTTP headers.

Just to help the old grey matter join the dots a bit more?